File: 20220817/모범답안/20220817_10.js/LeastSumOfKing.js
/**
 * @param arr {number[]}
 * @param K {number}
 * @returns {number}
 */
function solution(arr, K) {
  arr.sort((a, b) => a - b)

  const kings = []

  let i = 0
  while (i < arr.length) {
    let king = 0

    for (let sumCount = 0; sumCount < K; sumCount++, i++) {
      king = Math.max(king, arr[i])
    }

    kings.push(king)
  }

  return kings.reduce((pre, cur) => pre + cur, 0)
}

solution

File: 20220817/모범답안/20220817_10.js/LeastSumOfKing.py
from functools import reduce


def solution(arr, K):
    """
    :param arr: int[]
    :param K: int
    :return: int
    """

    arr.sort()

    kings = []

    i = 0
    while i < len(arr):
        king = 0

        for sumCount in range(K):
            king = max(king, arr[i])
            i += 1

        kings.append(king)

    return reduce(lambda pre, cur: pre + cur, kings, 0)
